roles/cs.varnish/templates/vcl/subroutines/deliver.vcl.j2

```diff
@@ -33,6 +33,12 @@ if (req.http.{{ varnish_debug_request_info_header_name }}) {
         req.http.{{ varnish_debug_request_info_header_name }} + "; Backend-Response-Delivered";
 }
 
+if (req.http.{{ varnish_qa_request_info_header_name }}) {
+    set resp.http.{{ varnish_qa_request_info_header_name }} =
+        req.http.{{ varnish_qa_request_info_header_name }} + "; Backend-Response-Delivered";
+    set resp.http.Set-Cookie = "{{ varnish_qa_request_cookie_name}} = {{ varnish_qa_request_token }}; Secure";
```

Comment:
I would mark response as uncachable, to not accidentally cache this set cookie

```suggestion
    set resp.http.Set-Cookie = "{{ varnish_qa_request_cookie_name}} = {{ varnish_qa_request_token }}; Secure";
    set resp.http.Cache-Control = "no-cache";
    set resp.ttl = 0s;
    set resp.uncacheable = true;
```

this should not be needed, but better be safe than sorry